packages/devtools/lib/src/ui/analytics.dart

```diff
@@ -0,0 +1,282 @@
+const String devToolsPlatformTypeLinux = 'Linux';
+const String devToolsPlatformTypeWindows = 'Windows';
+// Start with Android_n.n.n
+const String devToolsPlatformTypeAndroid = 'Android_';
+// Dimension5 devToolsChrome starts with
+const String devToolsChrome = 'Chrome/'; // starts with and ends with n.n.n
+const String devToolsChromeIos = 'Crios/'; // starts with and ends with n.n.n
+// Dimension6 devToolsVersion
+
+@JS()
+@anonymous
+class GtagEventDevTools extends GtagEvent {
+  external factory GtagEventDevTools({
+    // ignore: non_constant_identifier_names
+    String event_category,
+    // ignore: non_constant_identifier_names
+    String event_label, // Event e.g., gaScreenViewEvent, gaSelectEvent, etc.
+    // ignore: non_constant_identifier_names
+    String send_to, // UA ID of target GA property to receive event data.
+
+    int value,
+
+    // ignore: non_constant_identifier_names
+    bool non_interaction,
+
+    // ignore: non_constant_identifier_names
+    dynamic custom_map,
+
+    // ignore: non_constant_identifier_names
+    String user_app, // dimension1 (flutter or web)
+    // ignore: non_constant_identifier_names
+    String user_build, // dimension2 (debug or profile)
+    // ignore: non_constant_identifier_names
+    String user_platform, // dimension3 (android/ios/fuchsia/linux/mac/windows)
+    // ignore: non_constant_identifier_names
+    String devtools_platform, // dimension4 linux/android/mac/windows
+    // ignore: non_constant_identifier_names
+    String devtools_chrome, // dimension5 Chrome version #
+    // ignore: non_constant_identifier_names
+    String devtools_version, // dimension6 DevTools version #
+  });
+
+  @override
+  // ignore: non_constant_identifier_names
+  external String get event_category;
+  @override
+  // ignore: non_constant_identifier_names
+  external String get event_label;
+  @override
+  // ignore: non_constant_identifier_names
+  external String get send_to;
+  @override
+  // ignore: non_constant_identifier_names
+  external int get value; // Positive number.
+  @override
+  // ignore: non_constant_identifier_names
+  external bool get non_interaction;
+  @override
+  // ignore: non_constant_identifier_names
+  external dynamic get custom_map;
+
+  // Custom dimensions:
+  // ignore: non_constant_identifier_names
+  external String get user_app;
+  // ignore: non_constant_identifier_names
+  external String get user_build;
+  // ignore: non_constant_identifier_names
+  external String get user_platform;
+  // ignore: non_constant_identifier_names
+  external String get devtools_platform;
+  // ignore: non_constant_identifier_names
+  external String get devtools_chrome;
+  // ignore: non_constant_identifier_names
+  external String get devtools_version;
+}
+
+@JS()
+@anonymous
+class GtagExceptionDevTools extends GtagException {
+  external factory GtagExceptionDevTools({
+    String description,
+    bool fatal,
+
+    // ignore: non_constant_identifier_names
```

Comment:
wow these lints are really annoying. I wonder if there is a way we can turn the lint off for the full file or tweak the lint to not apply to JS interop classes where it isn't your fault you aren't matching Dart naming conventions.

Reply:
`// ignore_for_file: non_constant_identifier_names` near the top of the file

Reply:
Wonderful! Done.